crates/js-sys/src/lib.rs

```diff
@@ -2050,6 +2050,152 @@ extern "C" {
     pub fn set_prototype_of(target: &Object, prototype: &JsValue) -> bool;
 }
 
+// RegExp
+#[wasm_bindgen]
+extern {
+    #[derive(Clone, Debug)]
+    pub type RegExp;
+
+    /// The exec() method executes a search for a match in a specified
+    /// string. Returns a result array, or null.
+    ///
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec
+    #[wasm_bindgen(method)]
+    pub fn exec(this: &RegExp, text: &str) -> Option<Array>;
+
+    /// The flags property returns a string consisting of the flags of
+    /// the current regular expression object.
+    ///
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/flags
+    #[wasm_bindgen(method, getter)]
+    pub fn flags(this: &RegExp) -> JsString;
+
+    /// The global property indicates whether or not the "g" flag is
+    /// used with the regular expression. global is a read-only
+    /// property of an individual regular expression instance.
+    ///
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/global
+    #[wasm_bindgen(method, getter)]
+    pub fn global(this: &RegExp) -> bool;
+
+    /// The ignoreCase property indicates whether or not the "i" flag
+    /// is used with the regular expression. ignoreCase is a read-only
+    /// property of an individual regular expression instance.
+    ///
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/ignoreCase
+    #[wasm_bindgen(method, getter, js_name = ignoreCase)]
+    pub fn ignore_case(this: &RegExp) -> bool;
+
+    /// The non-standard input property is a static property of
+    /// regular expressions that contains the string against which a
+    /// regular expression is matched. RegExp.$_ is an alias for this
+    /// property.
+    ///
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/input
+    #[wasm_bindgen(static_method_of = RegExp, getter)]
+    pub fn input() -> JsString;
+
+    /// The lastIndex is a read/write integer property of regular
+    /// expression instances that specifies the index at which to
+    /// start the next match.
+    ///
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex
+    #[wasm_bindgen(method, getter, structural, js_name = lastIndex)]
+    pub fn last_index(this: &RegExp) -> u32;
+    #[wasm_bindgen(method, setter, structural, js_name = set_lastIndex)]
```

Comment:
I think this `js_name` may not be quite right? Also is `structural` required here to get this working?

Reply:
Ah, yes, I should have left a comment on this one.

At first I set `js_name = lastIndex` for the setter. But then I got a custom attribute panic with the message: "setters must start with `set_`." Simply adding it to the `js_name` solved the problem, and also made the test pass, so I left it at that. But it does look odd. What is the correct way to do it?

As for `structural`, I must admit I don't fully grok the documentation for it. But leaving it out gives the error `Error: descriptor for id='lastIndex' not found`. MDN documents this one method as `regexp.lastIndex` rather than `RegExp.prototype.lastIndex`, and so I thought `structural` causes it to not go through the property. But that was mostly a guess.

Reply:
And just to be clear, the actual rust name always started with `set_` while I was facing that problem.

Reply:
Ah ok the `structural` definitely makes sense to avoid the prototype (that's the attribute's purpose), but I think the `js_name` here needs to be `lastIndex` because otherwise this will set the `set_lastIndex` property on the object

Reply:
Alright. Since `js_name = lastIndex` panics I'll remove `last_index` from this PR for now. I can open a new issue for the panic.